code/game/objects/random/random.dm

```diff
@@ -1123,4 +1123,18 @@ var/list/multi_point_spawns
 				/obj/item/clothing/suit/space/void/mining/alt,
 				/obj/item/clothing/head/helmet/space/void/mining/alt
 			)
-		)
\ No newline at end of file
+		)
+
+/obj/random/landmine
+	name = "Random Landmine"
+	desc = "This is a random landmine."
+	icon = 'icons/obj/weapons.dmi'
+	icon_state = "uglymine"
+	spawn_nothing_percentage = 25
+
+/obj/random/landmine/item_to_spawn()
+	return pick(prob(30);/obj/effect/mine,
+				prob(25);/obj/effect/mine/frag,
```

Comment:
This one's going to hurt a whole lot of people out there. Worst one to step on for sure.